docs/myriad-scheduler-architecture.md

```diff
@@ -0,0 +1,69 @@
+# Myriad Scheduler
+Myriad scheduler is comprised of components that interact with YARN and Mesos
+services. The following diagram outlines the main interactions.
+
+![scheduler architecture diagram](images/sched_arch_diagram.jpg)
+
+## Mesos Master Interactions
+* Myriad Scheduler
+
+  This is the entry point to register a framework scheduler with Mesos. Upon
+  registration, we start receiving Mesos offers.
+
+* Offer Feed
+
+  Mesos offers are accepted and stashed here.
+
+* Offer Lifecycle Manager
+
+  Interface to retrieve stashed offers, return unused offers back to Mesos.
+
+* Consumed Offer
+
+  Represents offers that have been consumed for a slave prior to scheduling run.
+  When offers are consumed, they increase the capacity of YARN Node Manager by
+  that amount. Note that no Mesos tasks are launched when offers are consumed.
+  They are launched only after the scheduling run.
+
+## Mesos Slave Interactions
+* Pseudo Task Manager
+
+  Interface to manage pseudo Mesos tasks that are launched to satisfy the container
+  allocations done by YARN scheduler.
+
+* Pseudo Task Resource Manager
+
+  Interface to manage the resources consumed by psuedo Mesos tasks and reuse them
+  as needed. Pseudo Mesos tasks are launched only when there isn't enough
+  resources from existing tasks available to satisfy YARN container allocations.
+  Similarly, when YARN containers complete, their resources are not immediately
+  released back to Mesos. They are reused for the next scheduling run. When
```

Comment:
YARN scheduler is triggered every few seconds. We hold on to an offer only until the next scheduling run. Once RM does the scheduling, we will return the unused offers. So I don't think we will have starvation problems.